### PR TITLE
Initial commit of wadm API scaffolding

### PR DESCRIPTION
**Pull request by autodidaddict**
This is, for the moment, a WIP. Once we like the API and shapes, then I can go through and make all the operations real.

### CODE REVIEW

File: wadm/lib/wadm/api/api_server.ex [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  end
+
+  defp handle_request({"app", "del"}, body) do
+    with {:ok, _request} <- Jason.decode(body) do
+      # TODO - remove from store
+      # %{
+      # "name" => "name",
+      # "versions" => []
+      # }
+      # if the list of versions is empty, _all_ versions will be removed
```
Comment (thomastaylor312) [inline wadm/lib/wadm/api/api_server.ex:83]: Are we going to support running multiple versions of the same app model?

File: wadm/lib/wadm/api/api_server.ex [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          ]
+        }
+      ]
+      |> Jason.encode!()
+
+    {:reply, response}
+  end
+
+  defp handle_request({"app", "put"}, body) do
+    # input here is a JSON version of the OAM model
```
Comment (thomastaylor312) [inline wadm/lib/wadm/api/api_server.ex:54]: Just to double check, is OAM what we want to use? Is that going to be too much overhead for what we need wadm to do?

**Comment by thomastaylor312** [comment]
Initial structure looks pretty good so far. No major concerns with the data shape right now

File: wadm/test/model/yaml_json_test.exs [hunk 3]
```diff
@@ -1,18 +1,26 @@
-defmodule WadmTest.Model.YamlTest do
+defmodule WadmTest.Model.YamlJsonTest do
```
Comment (brooksmtownsend) [inline wadm/test/model/yaml_json_test.exs:1]: integration tests from an initial commit? 😍

File: wadm/lib/wadm/api/api_server.ex [hunk 4]
```diff
@@ -0,0 +1,174 @@
+defmodule Wadm.Api.ApiServer do
+  @moduledoc """
+  Exposes the API for interacting with wadm over the NATS subject wadm.api.>
+  The following is a list of the operations available in this API (minus the prefix):
+
+  * `apps.get` - Retrieves a summary of the application models stored by this wadm cluster. This will include status information
+  * `app.put` - Adds a new application model
+  * `app.del` - Deletes an application model
+  * `app.deploy` - Deploys an application to a given lattice
+  * `app.undeploy` - Stops managing an application in the given lattice. Non-destructive.
```
[the remark this answers is not in the document]
Comment (autodidaddict) [reply wadm/lib/wadm/api/api_server.ex:10]: yeah I'm thinking of adding a `destructive` option to the `undeploy` command, that will, if set, terminate all the resources that were being managed by that deployment.

File: wadm/lib/wadm/api/api_server.ex (hunk 1 above)
Comment (autodidaddict) [reply wadm/lib/wadm/api/api_server.ex:83]: Only a single version of a particular spec will be deployed in any given lattice at one time.

Comment (thomastaylor312) [reply wadm/lib/wadm/api/api_server.ex:83]: Ok, so I read this comment as us supporting multiple versions. We may want to clarify

**Pull request merged**